cmd/iam-object-store.go

```diff
@@ -474,14 +475,30 @@ func (iamOS *IAMObjectStore) loadAll(ctx context.Context, sys *IAMSys) error {
 	}
 
 	// purge any expired entries which became expired now.
+	var expiredEntries []string
 	for k, v := range sys.iamUsersMap {
 		if v.IsExpired() {
 			delete(sys.iamUsersMap, k)
 			delete(sys.iamUserPolicyMap, k)
+			expiredEntries = append(expiredEntries, k)
+			if v.ParentUser != "" {
+				expiredEntries = append(expiredEntries, v.ParentUser)
```

Comment:
I did not understand the logic here, why you assume that a parent user is expired if a sub-user is expired ?

Reply:
Because ParentUser is same as sub-user in OpenID.

Reply:
It's a good catch, I think I don't need to use ParentUser here..